### PR TITLE
Filter Chats by user name

### LINKED ISSUE #204: Chat search keywords.
Keywords to be supported:
- by:/u:
  - author of the message (support id?)

Shouldn't be the case on searches on the 'People' tab.

### PR DESCRIPTION
**Pull request by Sashank999**
Closes #204 .

### CODE REVIEW

File: extension/scripts/features/searchchat/ttSearchChat.js [hunk 1]
```diff
@@ -29,8 +29,8 @@
 			const input = parent.find(".tt-chat-filter input");
 			if (!input) return;
 
-			const keyword = input.value;
-			if (keyword) searchChat(message, keyword);
+			const inputValue = input.value;
```
Comment (DeKleineKobini) [inline extension/scripts/features/searchchat/ttSearchChat.js:32]: Why would you change from a good variable name to a mediocre one?

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]
This is also a change that should probably be in the changelog as a change.

File: extension/scripts/features/searchchat/ttSearchChat.js (hunk 1 above)
Comment (Sashank999) [reply extension/scripts/features/searchchat/ttSearchChat.js:32]: The name ``keyword`` doesn't actually imply its use.
Whereas the name ``inputValue`` says that it is the value of the input.

Comment (DeKleineKobini) [reply extension/scripts/features/searchchat/ttSearchChat.js:32]: It's the keyword of the search, so it implies its use very well?

Comment (DeKleineKobini) [reply extension/scripts/features/searchchat/ttSearchChat.js:32]: If it was new code, I wouldn't mind. But it's already a good name, and imo this only makes it worse. 

Sure, it's the value of the input, but what does that really tell me? It's the keyword that the user want to search on.

Comment (Sashank999) [reply extension/scripts/features/searchchat/ttSearchChat.js]: I agree.

File: extension/scripts/features/searchchat/ttSearchChat.js [hunk 2]
```diff
@@ -136,8 +136,21 @@
...
+		let keyword, user, id;
+		if (inputValue.startsWith("by:") || inputValue.startsWith("u:")) {
+			const splitInput = inputValue.split(" ");
+			user = splitInput.shift().split(":")[1];
+			if (!isNaN(user)) id = user;
+			keyword = splitInput.join(" ");
+		}
+		const userName = message.find("a");
+		const messageText = message.find("span").innerText.toLowerCase();
+		if (id && !userName.href.includes(id)) {
```
Comment (DeKleineKobini) [inline extension/scripts/features/searchchat/ttSearchChat.js:149]: If it's an id, it should probably only match the exact id. I'll work on that.

**Pull request merged**